src/store/flow/block/set-contact-property.ts

```diff
@@ -0,0 +1,76 @@
+import {IRootState} from '@/store'
+import {findBlockOnActiveFlowWith as findBlock, IBlock, IContext, SetContactProperty} from '@floip/flow-runner'
+import {ActionTree, MutationTree} from 'vuex'
+import {IFlowsState} from '../index'
+
+export interface ISetContactPropertyIndex {
+  blockId: IBlock['uuid'],
+  key: SetContactProperty['property_key'],
+}
+
+export interface ISetContactPropertyIndexWithValue extends ISetContactPropertyIndex {
+  value: SetContactProperty['property_value'],
+}
+
+export const mutations: MutationTree<IFlowsState> = {
+  block_addContactProperty(state, {blockId, key, value}: ISetContactPropertyIndexWithValue) {
+    const block = findBlock(blockId, state as unknown as IContext)
+
+    // Ensure the block has a set_contact_property[]
+    if (!block.config.set_contact_property) {
+      block.config.set_contact_property = []
+    }
+
+    // We should avoid duplicates
+    if (block.config.set_contact_property.find(({property_key}) => property_key === key)) {
+      throw new Error(`set_contact_property[${key}] already exists`)
+    }
+
+    block.config.set_contact_property = [
+      {
+        property_key: key,
+        property_value: value,
+      },
+      ...block.config.set_contact_property ?? [],
```

Comment:
`set_contact_property` is already ensured to be defined. Would you like to remove one of them?

Reply:
Quick question: we're not `push`ing to the array because we want Vue reactivity, right? (and pushing would merely mutate the existing array)

Reply:
Bonus question: would it be more 'intuitive' to add the new element at the end of the array?

Reply:
Although it _should_ be defined when we invoke this action, it may not be in case of errors because it's not required according to typings:

```ts
export interface IBlockConfig extends Record<string, any> {
  set_contact_property?: SetContactProperty[]
}
```

Unless you insist, I would be more comfortable keeping the extra fallback instead of `block.config!.set_contact_property` 😉 

We have to prepend newly added properties to make them appear at all, this will have to change if we decide to show/edit all properties; [VMO-6509](https://viamoinc.atlassian.net/browse/VMO-6509).